packages/devtools-reps/src/object-inspector/components/ObjectInspector.js

```diff
@@ -130,25 +147,58 @@ class ObjectInspector extends Component<Props> {
     const { loadedProperties } = this.props;
     const { cachedNodes } = this;
 
-    return getChildren({
+    const children = getChildren({
       loadedProperties,
       cachedNodes,
       item
     });
+    if (Array.isArray(children)) {
+      return children.map(i => this.getEvaluatedItem(i));
+    }
+
+    if (children) {
+      return this.getEvaluatedItem(children);
+    }
```

Comment:
this still seems a bit complicated...

Reply:
The alternative would be to pass `evaluations` to `getChildren`, and pipe it through all the different function that can create nodes, which I find more complex than that.

Can you think of anything else that would make this more simple?

Reply:
moved to node utils